gameMechanics.js

```diff
@@ -0,0 +1,303 @@
+	}
+	return newSum;
+}
+
+var checkLowerLeftDiag = function(currentGrid, selectedColumn, currentRow, currentPlayer, currentSum) {
+	var newSum = currentSum;
+    var height = currentGrid[selectedColumn].length;
+    var width = currentGrid.length;
+	if ((currentRow == height - 1) || (selectedColumn == 0)) {
+		return newSum;
+	}
+	var newRow = currentRow + 1;
+	var newColumn = selectedColumn - 1;
+	if ((newRow < height) && (newColumn >= 0) && (currentGrid[newColumn][newRow] == checkPlayer(currentPlayer))) {
+		newSum++;
+		return checkLowerLeftDiag(currentGrid, newColumn, newRow, currentPlayer, newSum);
+	}
+	return newSum;
+}
+
+var checkUpperLeftDiag = function(currentGrid, selectedColumn, currentRow, currentPlayer, currentSum) {
+	var newSum = currentSum;
+    var height = currentGrid[selectedColumn].length;
+    var width = currentGrid.length;
+	if ((currentRow == 0) || (selectedColumn == 0)) {
+		return newSum;
+	}
+	var newRow = currentRow - 1;
+	var newColumn = selectedColumn - 1;
+	if ((newRow < height) && (newColumn >= 0) && (currentGrid[newColumn][newRow] == checkPlayer(currentPlayer))) {
+		newSum++;
+		return checkUpperLeftDiag(currentGrid, newColumn, newRow, currentPlayer, newSum);
+	}
+	return newSum;
+}
+
+var checkLowerRightDiag = function(currentGrid, selectedColumn, currentRow, currentPlayer, currentSum) {
+	var newSum = currentSum;
+    var height = currentGrid[selectedColumn].length;
+    var width = currentGrid.length;
+	if ((currentRow == height - 1) || (selectedColumn == width - 1)) {
+		return newSum;
+	}
+	var newRow = currentRow + 1;
+	var newColumn = selectedColumn + 1;
+	if ((newRow < height) && (newColumn < width) && (currentGrid[newColumn][newRow] == checkPlayer(currentPlayer))) {
+		newSum++;
+		return checkLowerRightDiag(currentGrid, newColumn, newRow, currentPlayer, newSum);
+	}
+	return newSum;
+}
+
+var checkUpperRightDiag = function(currentGrid, selectedColumn, currentRow, currentPlayer, currentSum) {
+	var newSum = currentSum;
+    var height = currentGrid[selectedColumn].length;
+    var width = currentGrid.length;
+	if ((currentRow == 0) || (selectedColumn == width - 1)) {
+		return newSum;
+	}
+	var newRow = currentRow - 1;
+	var newColumn = selectedColumn + 1;
+	if ((newRow >= 0) && (newColumn < width) && (currentGrid[newColumn][newRow] == checkPlayer(currentPlayer))) {
+		newSum++;
+		return checkUpperRightDiag(currentGrid, newColumn, newRow, currentPlayer, newSum);
+	}
+	return newSum;
+}
+
+var checkVertical = function(currentGrid, selectedColumn, currentRow, currentPlayer, currentSum) {
+	var newSum = currentSum;
+    var height = currentGrid[selectedColumn].length;
+	if (currentRow == height - 1) return currentSum;
+	var newRow = currentRow + 1;
+	if ((newRow < height) && (currentGrid[selectedColumn][newRow] == checkPlayer(currentPlayer))) {
+		newSum++;
+		return checkVertical(currentGrid, selectedColumn, newRow, currentPlayer, newSum);
+	}
+	return newSum;
+}
```

Comment:
These `checkSomething` functions look like they're packed with logic and conditionals - excellent candidates for unit testing. ;)